### PR TITLE
Add support for all Android versions; disabled only PATCH methods on unsupported

### PR DESCRIPTION
**Pull request by Katsute**
### Prerequisites
*If **all** checks are not passed then the pull request will be closed.*
- [x] I have checked that no other similar pull request already exists.
- [x] My code follows the general code style as the rest of the code.
- [x] I have checked that no sensitive information is exposed.
- [x] Build compiles.
- [x] Build passes test cases.
- [x] I have added/modified documentation (check if not applicable).
- [x] I have added/modified test cases (check if not applicable).

**Workflow Run:** *The link to the MyAnimeList CI workflow run for your branch. Check README for guide on remote tests.*
https://github.com/Katsute/Mal4J/actions/runs/1088587837

### Changes Made
*List any changes made and/or other relevant issues.*

 - Switches to Logger for logging instead of sys out
 - Adds logger to module-info
 - Remove exceptions preventing unsupported Android versions from running
 - Disable only `PATCH` methods on unsupported Android versions

tldr: only `PATCH` methods disabled if Android unsupported, everything else enabled.

### CODE REVIEW

**Comment by Katsute** [comment]
@Sinon-bot approve Passes local testing
